### PR TITLE
chore: delete the Aspect CLI plugin

### PR DESCRIPTION
**Pull request by alexeagle**
With that code removed, we're okay with licensing this as Apache 2.0

This assumes we'll add the `lint` command in the aspect-cli repo as a built-in command immediately.

### CODE REVIEW

File: docs/linting.md [hunk 1]
```diff
@@ -21,10 +21,8 @@ If you use the Aspect CLI, then include a block like the following in `.aspect/c
 
 ```yaml
 plugins:
```
Comment (gregmagolan) [inline docs/linting.md:23]: remove this bit about plugins entirely?

Comment (gregmagolan) [reply docs/linting.md:23]: ohhh. you meant this to be
```
lint:
  aspects:
    - //tools:lint.bzl%eslint
```

File: lint/BUILD.bazel [hunk 2]
```diff
@@ -21,7 +21,6 @@ bzl_library(
 
 bzl_library(
     name = "lint_test",
```
Comment (gregmagolan) [inline lint/BUILD.bazel:23]: bzl_library with no srcs?

**Review by gregmagolan — APPROVED** [review]
🦖

File: lint/BUILD.bazel (hunk 2 above)
Comment (alexeagle) [reply lint/BUILD.bazel:23]: I ran gazelle and it disappointed :/

**Pull request merged**